### ISSUE TITLE
Asynchronous Searches

### ISSUE THREAD

**Issue by fschiettecatte** [ga4gh-discovery/ga4gh-case-discovery#13]
* Do we want to support asynchronous searches? This would mean:
    * Allowing the `querier` to tell the `host` to run the search asynchronously.
    * Allowing the `host` to tell the `querier` that it decided to run the search asynchronously (or that the search can only be run asynchronously??).
    * In both cases a callback would need to be specified so that the `host` can return the results to the `querier`.
    * There needs to be a clear network path between the `querier` and the `host`.
    * All this would require the use of the `query identification` so that the `querier` can reconcile results with the original request.

**Comment by Relequestual** [comment]
What's the use case for allowing the Client (Querier) to tell the Server to run the search asynchronously?

> All this would require the use of the query identification so that the querier can reconcile results with the original request.

Maybe, maybe not. There are (at least) three possible ways to handle this:
- Client specifies an ID, which then gets used someway by the Server when it's ready to respond
- Server includes an ID that it assings to the request, while telling the Client it will be handled async (probably HTTP status 202)
- Client must specify a unique URL when allowing for async responses

The last option feels the least problematic to me, as we don't need to specify some way to use an ID, and in stead the Client is responsible for providing the callback URL.

How to handle situations?

The Server can require the use of a component which specifies the callback URL, either by default, or a response which denotes a client error (HTTP 422 in this case), allowing the client to fix the issue and retry. This approach is similar to content negotiation, and we would need to define a component which indicates instructions to the Client, like "essential missing component X", which allows the Client to include such a compoenent if they are able to do so, or report an error to the end user.

**Comment by Relequestual** [comment]
@harindra-a Do you know if this will be required for the initial implementations? I would guess not.

**Comment by fschiettecatte** [comment]
I would say that this is not required, it introduces a lot of complexity. We had this in MME v1 and tossed it out as not needed.

**Comment by harindra-a** [comment]
Agree, I don't see this being required and introduces complexity. Let's keep (at least) first release simple

**Comment by Relequestual** [comment]
Great, it will not be considered for the initial release.

We should watch out for anyone who requests this functionality, and ask them which of hte possible solutions they feel makes the most sense.
